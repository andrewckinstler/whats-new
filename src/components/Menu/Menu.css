nav {
  display: flex;
  flex-flow: column;
  padding-top: 20px;
  padding-left: 20px;
}

button {
  margin: 1em;
  background-color: #141527;
  color: #fff;
  border: none;
  font-size: 1.5em;
}

h1 {
  color: #fff;
}