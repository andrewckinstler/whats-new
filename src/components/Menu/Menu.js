import React from 'react';
import './Menu.css'

const Menu = ({ changeCategory, local, entertainment, health, science, technology }) => {
  return (
    <nav>
      <button onClick={() => changeCategory(local)}>Local News</button>
      <button onClick={() => changeCategory(entertainment)}>Entertainment</button>
      <button onClick={() => changeCategory(health)}>Health</button>
      <button onClick={() => changeCategory(science)}>Science</button>
      <button onClick={() => changeCategory(technology)}>Technology</button>
    </nav>
  )
}

export default Menu;