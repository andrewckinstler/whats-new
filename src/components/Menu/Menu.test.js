import React from 'react';
import ReactDOM from 'react-dom';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  it('should render a boilerplate of the UI', () => {
    const mockState = { news: 'MockNews' };
    const changeCategoryMock = jest.fn()


    const wrapper = shallow(<Menu
      currentCategory={mockState}
      changeCategory={changeCategoryMock}
      local={'local'}
      entertainment={'entertainment'}
      health={'health'}
      science={'science'}
      technology={'technology'}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should call the changeCategory prop when buttons are clicked', () => {
    const mockState = { news: 'MockNews' };
    const changeCategoryMock = jest.fn()


    const wrapper = shallow(<Menu
      currentCategory={mockState}
      changeCategory={changeCategoryMock}
      local={'local'}
      entertainment={'entertainment'}
      health={'health'}
      science={'science'}
      technology={'technology'}
    />)

    wrapper.find('.local').simulate('click');
    expect(changeCategoryMock).toHaveBeenCalledWith('local')
  })
})