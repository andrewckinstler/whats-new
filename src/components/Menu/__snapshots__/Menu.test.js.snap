// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render a boilerplate of the UI 1`] = `
<nav>
  <h1>
    What's New
  </h1>
  <button
    className="local"
    onClick={[Function]}
  >
    Local News
  </button>
  <button
    className="entertainment"
    onClick={[Function]}
  >
    Entertainment
  </button>
  <button
    className="health"
    onClick={[Function]}
  >
    Health
  </button>
  <button
    className="science"
    onClick={[Function]}
  >
    Science
  </button>
  <button
    className="tech"
    onClick={[Function]}
  >
    Technology
  </button>
</nav>
`;
